config/routes.rb

```diff
@@ -3,4 +3,11 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+
+  root "users#index"
+
+  get "/users" => "users#index"
+  get "/users/:author_id" => "posts#index", as: :user_posts
+  get "/users/:author_id/posts/:id" => "posts#show", as: :user_post
+  get "/users/:id" => "users#show", as: :user
```

Comment:
- [ ] Instead of defining our routes strenuously this way, why don't we use the `resources` helper as provided by rails as it would be more concise? And then we can expose the routes we need using the **`only`** keyword. We can also nest the **posts** into the **users** routes, like so:

  ```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
  ```

  ![routes-show](https://user-images.githubusercontent.com/36542986/178380507-c5d174ac-b653-439b-a8fd-c5f03d976ef6.jpg)